MetaMask/metamask-extension
# Security provider check (OpenSea)

## Explanation
The security provider is an optional feature for users, similar to an RPC provider for a new chain.  The security provider will be called before the wallet action is presented to the user, and can flag possible malicious transactions that the wallet provider can raise to the user. 
More information: https://www.notion.so/opensea/Wallet-Tx-Security-PoC-b6a926e1cb6448638d147696cd8f5937

* Fixes #16281

In order to use the security provider check, `Transaction security check` should be enabled from the `Experimental` tab.

This functionality will be fully implemented in the following PR (https://github.com/MetaMask/metamask-extension/issues/16258) where the warning message will be shown.

## Manual testing steps

1. Go to `Experimental` tab and enable the `Transaction security check`.
2. Go to `test dapp`.
3. Send a transaction or sign a message and the security provider check will be called.
